cms/djangoapps/models/settings/course_grading.py

```diff
@@ -66,10 +67,40 @@ def update_from_json(course_key, jsondict, user):
         Decode the json into CourseGradingModel and save any changes. Returns the modified model.
         Probably not the usual path for updates as it's too coarse grained.
         """
+
+        if material_recompute_only(course_key):
+            return CourseGradingModel.update_from_json_selective(course_key, jsondict, user)
+        else:
+            descriptor = modulestore().get_course(course_key)
+
+            graders_parsed = [CourseGradingModel.parse_grader(jsonele) for jsonele in jsondict['graders']]
+            descriptor.raw_grader = graders_parsed
+            descriptor.grade_cutoffs = jsondict['grade_cutoffs']
+
+            modulestore().update_item(descriptor, user.id)
+
+            CourseGradingModel.update_grace_period_from_json(course_key, jsondict['grace_period'], user)
+
+            CourseGradingModel.update_minimum_grade_credit_from_json(course_key, jsondict['minimum_grade_credit'], user)
+            _grading_event_and_signal(course_key, user.id)
+
+            return CourseGradingModel.fetch(course_key)
+
+    @staticmethod
+    def update_from_json_selective(course_key, jsondict, user):
```

Comment:
This is almost completely copy and pasted from the above method, I think it would be nicer as just changes to the original method:
- remove the if
- add the fire_signal variable
- move the waffle check to the must_fire method and return true if the waffle flag isn't set

Reply:
The reason I decided to do it that way is in the hope that the waffle flag will be removed soon this will make it for easier reconsolidation.